sdk/storage/azure_storage_blob/tests/blob_container_client.rs

```diff
@@ -82,3 +83,159 @@ async fn test_set_container_metadata(ctx: TestContext) -> Result<(), Box<dyn Err
+        assert!(blob_names.contains(&blob_name));
+        assert_eq!(BlobType::BlockBlob, blob_type);
+    }
+
+    // Second Page
+    let list_blobs_options = BlobContainerClientListBlobFlatSegmentOptions {
+        marker: continuation_token,
+        ..Default::default()
+    };
+    let mut list_blobs_response = container_client.list_blobs(Some(list_blobs_options))?;
+    let second_page = list_blobs_response.try_next().await?;
+    let list_blob_segment_response = second_page.unwrap().into_body().await?;
+    let blob_list = list_blob_segment_response.segment.blob_items;
+    assert_eq!(2, blob_list.len());
+    for blob in blob_list {
+        let blob_name = blob.name.unwrap().content.unwrap();
+        println!("2nd page: {}", blob_name.clone());
+        let blob_type = blob.properties.unwrap().blob_type.unwrap();
+        assert!(blob_names.contains(&blob_name));
+        assert_eq!(BlobType::BlockBlob, blob_type);
+    }
+
+    container_client.delete_container(None).await?;
+    Ok(())
+}
+
+#[recorded::test]
+async fn test_list_blobs_with_continuation_next(ctx: TestContext) -> Result<(), Box<dyn Error>> {
+    // Recording Setup
+
+    let recording = ctx.recording();
+    let container_client = get_container_client(recording, false).await?;
+    let blob_names = [
+        "testblob1".to_string(),
+        "testblob2".to_string(),
+        "testblob3".to_string(),
+        "testblob4".to_string(),
+    ];
+
+    container_client.create_container(None).await?;
+    create_test_blob(&container_client.blob_client(blob_names[0].clone())).await?;
+    create_test_blob(&container_client.blob_client(blob_names[1].clone())).await?;
+    create_test_blob(&container_client.blob_client(blob_names[2].clone())).await?;
+    create_test_blob(&container_client.blob_client(blob_names[3].clone())).await?;
+
+    let list_blobs_options = BlobContainerClientListBlobFlatSegmentOptions {
+        maxresults: Some(2),
+        ..Default::default()
+    };
+    let mut pager_response = container_client.list_blobs(Some(list_blobs_options))?;
+    let mut page_count = 0;
+
+    while let Some(page) = pager_response.next().await {
+        page_count += 1;
+        let current_page = page.unwrap().into_body().await?;
+        match page_count {
+            1 => {
+                let blob_list = current_page.segment.blob_items;
+                assert_eq!(2, blob_list.len());
+
+                let mut first_loop_counter = 0;
+                for blob in blob_list {
+                    first_loop_counter += 1;
+                    println!("First loop: {}", first_loop_counter);
+                    let blob_name = blob.name.unwrap().content.unwrap();
+                    let blob_type = blob.properties.unwrap().blob_type.unwrap();
+                    assert!(blob_names.contains(&blob_name));
+                    assert_eq!(BlobType::BlockBlob, blob_type);
+                }
+            }
```

Comment:
Here is the relevant codeblock for the simple counter stdout. For completeness, stdout would read this after L218:
>First loop: 1
First loop: 2
First loop: 1
First loop: 2